web/static/js/components/layout/HeaderContainer.jsx

```diff
@@ -47,42 +47,78 @@ HeaderContainer.propTypes = {
   tabs: PropTypes.node,
   logout: PropTypes.func.isRequired,
   user: PropTypes.string.isRequired,
-  surveyFolder: PropTypes.object,
+  folder: PropTypes.object,
   panelSurveyId: PropTypes.number,
-  panelSurvey: PropTypes.object
+  panelSurveyFromOccurrence: PropTypes.object
 }
 
-const folder = (params, state) => {
-  const { surveyId, panelSurveyId } = params
-  let folderId = null
-  if (state.survey.data && state.survey.data.id == surveyId) {
-    folderId = state.survey.data.folderId
-  } else if (state.panelSurvey.data && state.panelSurvey.data.id == panelSurveyId) {
-    folderId = state.panelSurvey.data.folderId
+const getSurveyFromParams = (params, state) => {
+  const surveyId = params.surveyId && parseInt(params.surveyId)
+  if (!surveyId) return null
+  const survey = state.survey && state.survey.data
+  if (!survey) return null
+  if (survey.id == surveyId) {
+    return survey
+  } else {
+    return null
   }
-  const folder = state.folder && state.folder.folders && state.folder.folders[folderId]
-  return folder || null
 }
 
-const panelSurvey = (surveyId, state) => {
-  if (surveyId && state.survey && state.survey.data && state.survey.data.id == surveyId) {
-    const survey = state.survey.data
-    if (state.panelSurveys.items && state.panelSurveys.items[survey.panelSurveyOf]) {
-      return state.panelSurveys.items[survey.panelSurveyOf]
+const getPanelSurveyFromOccurrence = (survey, state) => {
+  if (survey) {
+    const panelSurvey = state.panelSurveys.items && state.panelSurveys.items[survey.panelSurveyId]
+    if (panelSurvey) {
+      return panelSurvey
     }
   }
   return null
 }
 
+const getPanelSurveyFromParams = (params, state) => {
+  const panelSurveyId = params.panelSurveyId && parseInt(params.panelSurveyId)
+  if (!panelSurveyId) return null
+  const panelSurvey = state.panelSurvey && state.panelSurvey.data
+  if (!panelSurvey) return null
+  if (panelSurvey.id == panelSurveyId) {
+    return panelSurvey
+  } else {
+    return null
+  }
+}
```

Comment:
this is the exact same logic as `getSurveyFromParams` but only differs which params to extract from `params` and from `state`.. perhaps it can be unified?